pkg/rancher-desktop/components/NetworkStatus.vue

```diff
@@ -0,0 +1,107 @@
+<script>
+import Vue from 'vue';
+
+import { ipcRenderer } from '@pkg/utils/ipcRenderer';
+import { networkStatus } from '@pkg/utils/networks';
+
+export default Vue.extend({
+  props: {
+    icon: {
+      type:    String,
+      default: '',
+    },
+    isStatusBarItem: {
+      type:    Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return { networkStatus: true };
+  },
+  computed: {
+    networkStatusLabel() {
+      return this.networkStatus ? networkStatus.CONNECTED : networkStatus.OFFLINE;
+    },
+    getTooltip() {
+      return {
+        content:   `<b>${ this.t('product.networkStatus') }</b>: ${ this.networkStatusLabel }`,
+        placement: 'top',
+        classes:   'tooltip-footer',
+      };
+    },
+  },
+  mounted() {
+    this.onNetworkStatusUpdate(window.navigator.onLine);
+    ipcRenderer.on('update-network-status', (event, status) => {
+      this.onNetworkStatusUpdate(status);
+    });
+    window.addEventListener('online', () => {
+      this.onNetworkStatusUpdate(true);
+    });
+    window.addEventListener('offline', () => {
+      this.onNetworkStatusUpdate(false);
+    });
+    // This event is triggered when the Preferences page is revealed (among other times).
+    // If the network status changed while the window was closed, this will update it.
+    window.addEventListener('pageshow', () => {
+      this.onNetworkStatusUpdate(window.navigator.onLine);
+    });
+  },
+  methods: {
+    onNetworkStatusUpdate(status) {
+      this.$data.networkStatus = status;
+    },
+  },
+});
+</script>
+
+<template>
+  <span
+    v-tooltip="isStatusBarItem ? getTooltip : {}"
+    class="networkStatusInfo"
+  >
+    <i
+      v-if="icon"
+      class="item-icon"
```

Comment:
We might want to consider if we can just change the color of the icon to red/green instead of having the dot 🤔 

![image](https://github.com/rancher-sandbox/rancher-desktop/assets/835961/1ceb0feb-463a-490a-bacd-c0bdcefa194d)

I don't think that we want to change anything for this PR, but we might want to raise the idea during a design discussion.

Reply:
Oh... yes, indeed. @manuelecarlini and I talked already about it. There are two possible downsides:

1. If the icon is not shown, we don't have a colored indicator
2. It might work well on the dark theme. I'm not too sure about the light one. Maybe with the globe icon it's better than with the placeholder icon before.

The green color code is not final, yet. Once it is, I'll try it out and prepare some screenshots for discussion.

Reply:
We could also make the word `online` green and `offline` red.

Whatever we do, colour should not be the only way we specify the value, so people with colour-vision impairment can still figure out the status without having to go to the hover tip.

Reply:
I created an issue for further discussion here: https://github.com/rancher-sandbox/rancher-desktop/issues/6046